source/include/Synchronization.h

```diff
@@ -69,12 +69,15 @@ class OccurrenceCore : public ObservableCore
 };
 typedef TypedObject<OccurrenceCore> OccurrenceObject, *OccurrenceRef;
 
+const Int32 kMaxExecWakeUpTime = 10000;  // (milliseconds).  10 seconds.
+
 //------------------------------------------------------------
 //! Timer object that clumps can wait on.
 class Timer : public ObservableCore
 {
  public:
     Boolean AnythingWaiting()                   { return _observerList != null; }
+    IntMax NextWakeUpTime()                     { return _observerList != null ? _observerList->_info : 0; }
```

Comment:
By looking at the comment where _info is declared, _info contains state information, but in this case it appears we expect it to contain time information, is this correct?

Reply:
Yes.  The _info field is used differently by different inherited classes of ObservableCore. 
For Timers, it's always a wake up timer value.  (For Occurrences, it's the next occurrence 'count'.  I didn't look at all of them.)